src/species_advance/species_advance.cc

```diff
@@ -44,13 +45,17 @@ restore_species( void )
   RESTORE( sp );
   RESTORE_STR( sp->name );
   sp->p  = (particle_t *)        restore_data();
-  #ifdef VPIC_GLOBAL_PARTICLE_ID
-  sp->p_id  = (size_t*)          restore_data();
-  #endif
   sp->pm = (particle_mover_t *)  restore_data();
   RESTORE_ALIGNED( sp->partition );
   RESTORE_PTR( sp->g );
   RESTORE_PTR( sp->next );
+  #ifdef VPIC_GLOBAL_PARTICLE_ID
+  if(sp->has_ids) {
+    sp->p_id  = (size_t*)        restore_data();
+  } else {
+    sp->p_id  = (size_t*)        NULL;
```

Comment:
It is initially nullprt (as per line 168), and I hope the restore_data() inside the RESTORE() macro would recreate that. But in case of `has_ids` I think we need the explicit call to restore_data.

Reply:
OK. Just to check: have you thought about the semantics of `NULL` vs `nullptr`? I need to think about it more, but my gut says we might prefer `nullptr` in most cases here?